Daviddt17/island-android-app
# Save outgoing events and push events from sync adapter

This PR hardens the protocol by handling situations when network is not available. The event publisher now saves events to the content provider. The sync service reads the events from the content provider and only deletes them if they successfully post to service.

To test this
- connect to wireless at LCCC but don't authenticate with the service
- make a post, it will result in a IO exception (on master branch this post will be lost forever and it will be impossible for friends to get any more content from this user)
- connect to network or turn off wifi
- make another post or do a swipe refresh. This should push the event that previously failed
